### ISSUE TITLE
Wildcard search 

### ISSUE THREAD

**Issue by timmrk** [pear/Services_Openstreetmap#22]
In the current state you can just search for exact key value pairs in the loaded XML/data.
For example: "highway => residential"  or "country => FR" in the search-Method.

I dont know if the OSM API has some kind of wildcard search already. But i do know that the Overpass/Turbo API has a Wildcard Search in the form of "key=*" .  [Overpass Tag Search](https://wiki.openstreetmap.org/wiki/Overpass_turbo/Wizard#Tag_Searches).

It would be nice if you can perform an wildcard search, like if you want to get all highway-types from an loaded Area.

**Comment by kenguest** [comment]
@timmrk does the fix that I've committed in 1c26666 work the way you would like?

**Comment by timmrk** [comment]
Hello,

yes it works as i thought it should.
My Code for Using it in my Programm looks like this:


``` php
$results = $osm->search(array("highway" => "*"));
foreach ($results as $result) {
      $name = $result->getTag('name');
      if (!in_array($name, $resultarray)){
          $resultarray[]=$name;
      }
}
```
This way it give my for example all streetnames of all streettypes back. Additionally in the Code i check if the Name exists in the array, to not have duplicates. Thank you!
